File: week-01/p1.go
package main

import "fmt"

func findMaxValue(l []int) int {
	maxV := l[0]
	for _, v := range l {
		if v > maxV {
			maxV = v
		}
	}
	return maxV
}

func findSecondMaxValue(l []int) int {
	maxV := findMaxValue(l)
	newL := []int{}
	for _, v := range l {
		if v != maxV {
			newL = append(newL, v)
		}
	}
	return findMaxValue(newL)
}

func main() {
	a := []int{7, 6, 9, 3, 2, 5}
	fmt.Println(findMaxValue(a))
	fmt.Println(findSecondMaxValue(a))
}

File: week-01/p1.py
def find_max_value(l: list[int]) -> int:
    max_v: int = l[0]
    v: int
    for v in l:
        if v > max_v:
            max_v = v
    return max_v

def find_second_max_value(l: list[int]) -> int:
    max_v: int = find_max_value(l)
    new_l: list[int] = [v for v in l if v != max_v]
    return find_max_value(new_l)

def main():
    a: list[int] = [7, 6, 9, 3, 2, 5] 
    print(find_max_value(a))
    print(find_second_max_value(a))


if __name__ == "__main__":
    main()